app/code/local/Magmodules/Webwinkelconnect/Helper/Data.php

```diff
@@ -143,6 +143,81 @@ public function syncProductReview(array $product_review): array {
         }
     }
 
+
+    public function hasConsent(Mage_Sales_Model_Order $order): bool {
+        $permission_url = 'https://dashboard.webwinkelkeur.nl/api/2.0/order_permissions.json?' .
+            http_build_query([
+                'orderNumber' => $order->getIncrementId(),
+                'id' => $this->getShopId(),
+                'code' => $this->getApiKey(),
+            ]);
+
+        $curl = new Varien_Http_Adapter_Curl();
+        $curl->setConfig([
+            'timeout'   => 10,
+        ]);
+        $curl->addOptions([
+            CURLOPT_RETURNTRANSFER => true,
+            CURLOPT_FOLLOWLOCATION => true,
+            CURLOPT_FAILONERROR => true,
+        ]);
+        $curl->write(Zend_Http_Client::GET, $permission_url, '2', false);
+        $response = $curl->read();
```

Comment:
I suggested using `Varien_Http_Adapter_Curl` earlier, but at this point I think it would've been better not to use it. It makes some things a bit harder since by default it includes the headers in the response, and it also has some weird logic for dealing with responses that consist of multiple parts (see `Varien_Http_Adapter_Curl::read`).

In your code, for example, empty responses are treated as errors (even though they aren't), and the response code is extracted from the response string, which is unnecessary if you just use cURL directly without setting CURLOPT_HEADER. I have no idea why Magento does it this way, and I think we'd better just skip that strange wrapper.

So in conclusion, it sucks, and I suggest we don't use it after all.

Could you rewrite this code (and future code) to use cURL directly?